app/assets/javascripts/discourse/app/models/site.js

```diff
@@ -87,9 +87,19 @@ const Site = RestModel.extend({
   trackedCategoriesList(categories) {
     const trackedCategories = [];
 
+    const uncategorizedCategoryId = Site.currentProp(
+      "uncategorized_category_id"
+    );
+
     for (const category of categories) {
       if (category.isTracked) {
-        trackedCategories.push(category);
+        if (
+          !this.siteSettings.suppress_uncategorized_badge ||
+          (this.siteSettings.suppress_uncategorized_badge &&
+            category.id !== uncategorizedCategoryId)
```

Comment:
```suggestion
          !this.siteSettings.suppress_uncategorized_badge ||
          category.id !== uncategorizedCategoryId
```

I think it can be simplified to this (prettier may not like this code format).

Reply:
You're right here let me fix that.